homeassistant/components/hassio/update.py

```diff
@@ -0,0 +1,288 @@
+        try:
+            await async_update_os(self.hass, version)
+        except HassioAPIError as err:
+            raise HomeAssistantError(
+                f"Error updating Home Assistant Operating System: {err}"
+            ) from err
+
+
+class SupervisorSupervisorUpdateEntity(HassioSupervisorEntity, UpdateEntity):
+    """Update entity to handle updates for the Home Assistant Supervisor."""
+
+    _attr_supported_features = UpdateEntityFeature.INSTALL
+    _attr_title = "Home Assistant Supervisor"
+
+    @property
+    def latest_version(self) -> str:
+        """Return native value of entity."""
+        return self.coordinator.data[DATA_KEY_SUPERVISOR][ATTR_VERSION_LATEST]
+
+    @property
+    def current_version(self) -> str:
+        """Return native value of entity."""
+        return self.coordinator.data[DATA_KEY_SUPERVISOR][ATTR_VERSION]
+
+    @property
+    def release_url(self) -> str | None:
+        """URL to the full release notes of the latest version available."""
+        if (version := self.latest_version) is None:
+            return None
+        if "dev" in version:
+            return "https://github.com/home-assistant/supervisor/commits/main"
+        return f"https://github.com/home-assistant/supervisor/releases/tag/{version}"
+
+    @property
+    def entity_picture(self) -> str | None:
+        """Return the iconof the entity."""
+        return "https://brands.home-assistant.io/hassio/icon.png"
+
+    async def async_install(
+        self,
+        version: str | None = None,
+        backup: bool | None = None,
+        **kwargs: Any,
+    ) -> None:
+        """Install an update."""
+        try:
+            await async_update_supervisor(self.hass)
+        except HassioAPIError as err:
+            raise HomeAssistantError(
+                f"Error updating Home Assistant Supervisor: {err}"
+            ) from err
+
+
+class SupervisorCoreUpdateEntity(HassioCoreEntity, UpdateEntity):
+    """Update entity to handle updates for Home Assistant Core."""
+
+    _attr_supported_features = (
+        UpdateEntityFeature.INSTALL
+        | UpdateEntityFeature.SPECIFIC_VERSION
+        | UpdateEntityFeature.BACKUP
+    )
+    _attr_title = "Home Assistant Core"
+
+    @property
+    def latest_version(self) -> str:
+        """Return native value of entity."""
+        return self.coordinator.data[DATA_KEY_SUPERVISOR][ATTR_VERSION_LATEST]
+
+    @property
+    def current_version(self) -> str:
+        """Return native value of entity."""
+        return self.coordinator.data[DATA_KEY_SUPERVISOR][ATTR_VERSION]
+
+    @property
+    def entity_picture(self) -> str | None:
+        """Return the iconof the entity."""
+        return "https://brands.home-assistant.io/homeassistant/icon.png"
+
+    @property
+    def release_url(self) -> str | None:
+        """URL to the full release notes of the latest version available."""
+        if (version := self.latest_version) is None:
+            return None
```

Comment:
```suggestion
```

We can still show the latest version?

Reply:
Yeah, we can fall back to the "stable" latest release here.